cache/contenthash/checksum.go

```diff
@@ -91,6 +91,12 @@ type cacheManager struct {
 }
 
 func (cm *cacheManager) Checksum(ctx context.Context, ref cache.ImmutableRef, p string, opts ChecksumOpts, s session.Group) (digest.Digest, error) {
+	if ref == nil {
+		if p == "/" {
+			return digest.FromBytes(nil), nil
+		}
+		return "", errors.Errorf("cannot checksum empty reference")
```

Comment:
nit: probably better to return the same "not found" error that would happen if `p` does not exist. "empty reference" is an internal concept, users of build don't know what it means